jupyter_server/services/events/handlers.py

```diff
@@ -75,6 +79,45 @@ def on_close(self):
         self.event_bus.handlers.remove(self.logging_handler)
 
 
+class EventHandler(APIHandler):
+    """REST api handler for events"""
+
+    auth_resource = AUTH_RESOURCE
+
+    @property
+    def event_bus(self):
+        """Jupyter Server's event bus that emits structured event data."""
+        return self.settings["event_bus"]
+
+    @web.authenticated
+    @authorized
+    async def post(self):
+        payload = self.get_json_body()
+        if payload is None:
+            raise web.HTTPError(400, "No JSON data provided")
+
+        try:
+            if "timestamp" in payload:
+                timestamp = datetime.strptime(payload["timestamp"], "%Y-%m-%d %H:%M:%S")
+            else:
+                timestamp = None
+
+            self.event_bus.record_event(
+                schema_name=payload["schema_name"],
+                version=payload["version"],
+                event=payload["event"],
+                timestamp_override=timestamp,
+            )
+            self.set_status(201)
```

Comment:
Since we aren't creating a REST resource and aren't going to give the client a Location header or a payload with the newly created resource ... indeed since this isn't exactly a newly created resource, I would propose a different success code: either 200 (OK) or 202 (Accepted) or 204 (No Content).

Reply:
Thanks for reviewing. Updated in the latest push.